src/Reexport.jl

```diff
@@ -1,11 +1,21 @@
 module Reexport
 
-macro reexport(ex)
-    isa(ex, Expr) && (ex.head == :module ||
-                      ex.head == :using ||
-                      (ex.head == :toplevel &&
-                       all(e->isa(e, Expr) && e.head == :using, ex.args))) ||
-        error("@reexport: syntax error")
+macro reexport(ex::Expr)
+    esc(reexport(__module__, ex))
+end
+
+function reexport(m::Module, ex::Expr)
+    # unpack any macros
+    ex = macroexpand(m, ex)
+    # recursively unpack any blocks
+    if ex.head == :block
+        ex = Base.remove_linenums!(ex)
```

Comment:
Rather than removing line numbers, you can skip them during processing, which will preserve them in the transformed expression. One way to do this is to define
```julia
reexport(m::Module, linenum::LineNumberNode) = linenum
```
then remove the `remove_linenums!` and let the `map` below handle this case. Alternatively, you can remove `remove_linenums!` and change the `map` call to something like
```julia
map(e -> e isa LineNumberNode ? e : reexport(m, e), ex.args)
```

Reply:
Thank you for the suggestions, this is a good idea. I added the identity dispatch for `LineNumberNode` which seems most julin.